### ISSUE TITLE
Can't deploy w/ docker

### ISSUE THREAD

**Issue by eladeyal-intel** [openSUSE/osem#2906]
I'm deploying with docker-compose (fixed #2905), database is ready, but can't connect via web  browser.

```
started with pid 8
started with pid 9
=> Booting Puma
=> Rails 5.2.6 application starting in production
=> Run `rails server -h` for more startup options
[8] Puma starting in cluster mode...
[8] * Version 4.3.9 (ruby 2.5.9-p229), codename: Mysterious Traveller
[8] * Min threads: 5, max threads: 5
[8] * Environment: production
[8] * Process workers: 2
[8] * Preloading application
[8] * Listening on tcp://0.0.0.0:3000
[8] Use Ctrl-C to stop
[8] - Worker 0 (pid: 24) booted, phase: 0
[8] - Worker 1 (pid: 38) booted, phase: 0
2021-11-08 18:22:12 +0000: HTTP parse error, malformed request (): #<Puma::HttpParserError: Invalid HTTP format, parsing fails.>
```

And the client

```
curl -v -4 https://localhost:8080/
*   Trying 127.0.0.1...
* TCP_NODELAY set
* Connected to localhost (127.0.0.1) port 8080 (#0)
* ALPN, offering h2
* ALPN, offering http/1.1
* successfully set certificate verify locations:
*   CAfile: /etc/pki/tls/certs/ca-bundle.crt
  CApath: none
* TLSv1.3 (OUT), TLS handshake, Client hello (1):
* error:1408F10B:SSL routines:ssl3_get_record:wrong version number
* Closing connection 0
curl: (35) error:1408F10B:SSL routines:ssl3_get_record:wrong version number
```

Help appreciated

@hennevogel I see you just touched this recently

**Comment by eduardoj** [comment]
One of the lines of docker-compose output says:
`Listening on tcp://0.0.0.0:3000`.

Try with `curl -v -4 https://localhost:3000/`.

**Comment by hennevogel** [comment]
There is no SSL for the container, you will have to do this in your reverse proxy (apache, nginx or whatnot).

**Comment by eladeyal-intel** [comment]
`curl -v -4 http://localhost:8080` responds with 302 REDIRECT to the https address.

Is the production compose file working at all?

Thanks @hennevogel

**Comment by hennevogel** [comment]
You are right, there was a problem with the SSL environment variable handling. Fixed with #2907
